src/application_config/configuring_base.py

```diff
@@ -21,49 +22,45 @@
 
 
 @dataclass(frozen=True)
-class ConfigSectionBase(ABC):
-    """Base class for all ConfigSection classes"""
+class ConfigSectionBase:
+    """Base class for all ConfigSection classes (so that we can bound a TypeVar)"""
 
 
-# The next line has a type: ignore because MyPy has a problem with @abstractmethod
-# If you comment the line with @abstractmethod, you can do type checking
 @dataclass(frozen=True)
-class ConfigBase(ABC):
+class ConfigBase:
     """Base class for main Config class"""
 
-    @staticmethod
-    @abstractmethod
-    def get_app_basename() -> str:
-        """Return the application base name.
-
-        This name, with a preceding dot, will also be the folder name in the home directory
-        that will store the config.
-        """
+    @classmethod
+    def default_config_foldername(cls: type[TConfig]) -> str:
```

Comment:
I think we should return a `Path` here. A custom implementation might want to insert a number of subfolders here (e.g. `custom/path/to/config`)

Reply:
Maybe it should the also be `default_config_dir`, or `default_configdir`)

Reply:
I propose to leave this signature as is; we have a "default_config_filepath" that returns a path (including the filename) and that can be overridden to accomplish what you describe.